### DISCUSSION TITLE
Possible bug on extensions/control-plane/store/sql/transfer-process-store-sql/src/main/java/org/eclipse/dataspaceconnector/sql/transferprocess/store/schema/BaseSqlDialectStatements.java

### DISCUSSION THREAD

**Discussion by lucian-torje-siemens** [eclipse-edc/Connector#2059] [General]
Hello,
The select seems to use the process id instead of transfer id - is it a bug or was intentionally implemented like this (and why) ?

![image](https://user-images.githubusercontent.com/102717569/194550447-180e3e43-d72b-4520-85f5-100490458313.png)

**Comment by ndr-brt** [comment]
Hi, thanks for reporting, could you provide a test case or some steps to reproduce a failing behavior? It will simplify the review/fix.

**Reply by lucian-torje-siemens** [reply]
Hi @ndr-brt , I was able to reproduce it while starting data transfer by using product-edc 0.1.1 with postgresql support. Note that even if I've used the same connector as both consumer/producer (therefore the same database) the scenario should be visible for normal flow (separate connectors).

**Reply by ndr-brt** [reply]
That's good. Could you provide the steps needed to replicate the bug? Because we have an end to end test using postgresql and it would be great to be able to replicate it there.

**Reply by lucian-torje-siemens** [reply]
Hi @ndr-brt 
I've tested it like:

Environment:
- code from https://github.com/catenax-ng/product-edc/tree/0.1.1 which uses EDC https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/tree/658c5e31accf5f7f4b221e94478763fd30af7d85
- single database
- single control plane / data plane (calling itself)

Steps:
- run locally a control plane and data plane
- create asset/policy/contract definition
- negotiate contract and start transfer 

Expected result:
- transfer done

Current result:
- Failed in provider code due to missing asset data address - reason: the SQL query was using the process id and in the case consumer/producer the behavior is different - as consumer it does not have the data address, only as provider it has an attached data address 

![image](https://user-images.githubusercontent.com/102717569/195300346-72872750-1cc1-426b-845a-d718faa60790.png)
![image](https://user-images.githubusercontent.com/102717569/195300429-f4ada334-abde-4960-8eb6-0d0f1617be93.png)

**Reply by ndr-brt** [reply]
@lucian-torje-siemens as I said, the "single instance calling itself" is not a supported use case, could you please test this with two separated connector with database on two different schemas?
